### PR TITLE
VL-43_Add-hightlights-to-datepicker-dates_Dmytro-Holdobin

### PR DESCRIPTION
**Pull request by Explicit12**
> https://ilevel.atlassian.net/browse/VL-43

### CODE REVIEW

File: src/ui.form-calendar/composables/attrs.composable.js [hunk 1]
```diff
@@ -30,7 +30,9 @@ export default function useAttrs(props) {
     attrs[`${key}Attrs`] = getAttrs(key, { classes });
 
     // eslint-disable-next-line prettier/prettier
-    if (["timepickerInputHours", "timepickerInputMinutes", "timepickerInputSeconds"].includes(key)) {
+    if (
+      ["timepickerInputHours", "timepickerInputMinutes", "timepickerInputSeconds"].includes(key)
+    ) {
```
Comment (itsJohnnyGrid) [inline src/ui.form-calendar/composables/attrs.composable.js:35]: eslint applied, rollback formatting

File: src/ui.form-date-picker-range/configs/default.config.js [hunk 2]
```diff
@@ -44,6 +44,11 @@ export default /*tw*/ {
   periodDateYearList: "grid grid-cols-3 grid-rows-1 gap-0.5",
   periodDateQuarterList: "",
   periodDate: "w-full",
+  periodDateInRange: "bg-brand-100 text-brand-900 hover:!bg-brand-200 rounded-none",
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/configs/default.config.js:47]: Should be general key with same styles for button and specific keys for each case to aviod styling hell in future :)

File: src/ui.form-date-picker-range/index.vue [hunk 3]
```diff
@@ -812,7 +812,7 @@ function onClickPeriodButton(periodName) {
     period.value = PERIOD.year;
   }
 
-  selectDate(periodDateList.value.at(0));
+  // selectDate(periodDateList.value.at(0));
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/index.vue:815]: ???

File: src/ui.form-date-picker-range/index.vue [hunk 4]
```diff
@@ -885,7 +885,67 @@ function getPeriodButtonsClasses(periodName) {
...
+  const startDateInRangeIndex = periodDateList.value.findIndex((periodDate) => {
+    return localStart <= periodDate.endRange && localStart >= periodDate.startRange;
+  });
+
+  const endDateInRangeIndex = periodDateList.value.findIndex((periodDate) => {
+    return localEnd >= periodDate.startRange && localEnd <= periodDate.endRange;
+  });
+
+  const isStartDateInRangeIndex = startDateInRangeIndex !== -1;
+  const isEndDateInRangeIndex = endDateInRangeIndex !== -1;
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/index.vue:916]: use ~ and const no needed

File: src/ui.form-date-picker-range/configs/default.config.js (hunk 2 above)
Comment (Explicit12) [reply src/ui.form-date-picker-range/configs/default.config.js:47]: `periodDate` is general key. This one and others are specific keys.

File: src/ui.form-date-picker-range/index.vue (hunk 4 above)
Comment (Explicit12) [reply src/ui.form-date-picker-range/index.vue]: Its still better to use const and keep result rather than call those operators each time.

Comment (itsJohnnyGrid) [reply src/ui.form-date-picker-range/index.vue]: Maybe but it not hevy operation. So I'm not sure this optimisation is really needed. You just comparing 2 values.

Comment (Explicit12) [reply src/ui.form-date-picker-range/index.vue]: It's not significant but does no harm and code is more readable this way, imo.

File: src/ui.form-date-picker-range/configs/default.config.js [hunk 5]
```diff
@@ -45,24 +45,25 @@ export default /*tw*/ {
   periodDateQuarterList: "",
   periodDate: "w-full",
   periodDateInRange: "bg-brand-100 text-brand-900 hover:!bg-brand-200 rounded-none",
-  firstPeriodGridDate: "rounded-dynamic rounded-r-none bg-brand-200 text-brand-900 hover:bg-brand-200",
-  lastPeriodGridDate: "rounded-dynamic rounded-l-none bg-brand-200 text-brand-900 hover:bg-brand-200",
-  firstPeriodListDate: "rounded-dynamic rounded-b-none bg-brand-200 text-brand-900 hover:bg-brand-200",
-  lastPeriodListDate: "rounded-dynamic rounded-t-none bg-brand-200 text-brand-900 hover:bg-brand-200",
+  edgePeriodDate: "bg-brand-200 text-brand-900 hover:bg-brand-200",
+  firstPeriodGridDate: "rounded-dynamic rounded-r-none",
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/configs/default.config.js:49]: rounded-dynamic you can move also

File: src/ui.form-date-picker-range/index.vue [hunk 6]
```diff
@@ -183,6 +183,7 @@
             :error="inputRangeFromError"
             size="md"
             v-bind="rangeInputAttrs"
+            class="group/range-input-first"
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/index.vue:186]: not good, should be in key classes

File: src/ui.form-date-picker-range/index.vue [hunk 7]
```diff
@@ -193,6 +194,7 @@
             :error="inputRangeToError"
             size="md"
             v-bind="rangeInputAttrs"
+            class="group/range-input-last"
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/index.vue:197]: same

File: src/ui.form-calendar/configs/default.config.js [hunk 8]
```diff
@@ -20,12 +20,18 @@ export default /*tw*/ {
   inRangeFirstDay: "rounded-dynamic rounded-r-none bg-brand-200 text-brand-900 hover:bg-brand-200",
   inRangeLastDay: "rounded-dynamic rounded-l-none bg-brand-200 text-brand-900 hover:bg-brand-200",
   inRangeDay: "bg-brand-100 text-brand-900 hover:!bg-brand-200 rounded-none",
+  inRangeFirstMonth: "rounded-dynamic rounded-r-none bg-brand-200 text-brand-900 hover:bg-brand-200",
+  inRangeLastMonth: "rounded-dynamic rounded-l-none bg-brand-200 text-brand-900 hover:bg-brand-200",
+  inRangeMonth: "bg-brand-100 text-brand-900 hover:!bg-brand-200 rounded-none",
+  inRangeFirstYear: "rounded-dynamic rounded-r-none bg-brand-200 text-brand-900 hover:bg-brand-200",
+  inRangeLastYear: "rounded-dynamic rounded-l-none bg-brand-200 text-brand-900 hover:bg-brand-200",
```
[the remark this answers is not in the document]
Comment (itsJohnnyGrid) [reply src/ui.form-calendar/configs/default.config.js:27]: not moved

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**